yt_dlp/extractor/crunchyroll.py

```diff
@@ -807,30 +828,46 @@ def _real_extract(self, url):
         hardsub_preference = qualities(requested_hardsubs[::-1])
         requested_formats = self._configuration_arg('format') or ['adaptive_hls']
 
-        formats = []
+        available_formats = {}
         for stream_type, streams in get_streams('streams'):
             if stream_type not in requested_formats:
                 continue
             for stream in streams.values():
-                hardsub_lang = stream.get('hardsub_locale') or ''
-                if hardsub_lang.lower() not in requested_hardsubs:
-                    continue
-                format_id = join_nonempty(stream_type, format_field(stream, 'hardsub_locale', 'hardsub-%s'))
                 if not stream.get('url'):
                     continue
-                if stream_type.endswith('hls'):
+                hardsub_lang = stream.get('hardsub_locale') or ''
+                format_id = join_nonempty(stream_type, format_field(stream, 'hardsub_locale', 'hardsub-%s'))
+                available_formats[hardsub_lang] = (stream_type, format_id, hardsub_lang, stream['url'])
+
+        if '' in available_formats and 'all' not in requested_hardsubs:
+            full_format_langs = set(requested_hardsubs)
+            self.to_screen(
+                'To expand certain hardsub formats use "--extractor-args crunchyrollbeta:hardsub=<language_code>". <language_code> can also be "all". See the README for more info.',
+                only_once=True)
```

Comment:
```suggestion
            if not self._configuration_arg('hardsub'):
                self.to_screen(
                    'To get all formats of a hardsub language, use '
                    '"--extractor-args crunchyrollbeta:hardsub=<language_code or all>". '
                    'See https://github.com/yt-dlp/yt-dlp#crunchyrollbeta for more info',
                    only_once=True)
```
how about this?

Reply:
I like splitting the string, but @tejing1 argued it might be good to remind people even if the argument is set. See [this](https://github.com/yt-dlp/yt-dlp/pull/5041#discussion_r984067988).

Reply:
I don't use this extractor, but if I did, I would personally be annoyed by a message telling me to do what I'm already doing. Could just be a me thing though. Go with whichever you guys think is best

Reply:
Then I believe leaving it in is the right way as well.